File: java/src/tarehart/rlbot/planning/TacticsTelemetry.java
package tarehart.rlbot.planning;

import tarehart.rlbot.Bot;

import java.util.*;

public class TacticsTelemetry {

    private static Map<Integer, TacticalSituation> tacticalSituations = new HashMap<>();


    public static void set(TacticalSituation situation, int playerIndex) {
        tacticalSituations.put(playerIndex, situation);
    }

    public static void reset(int playerIndex) {
        tacticalSituations.remove(playerIndex);
    }

    public static Optional<TacticalSituation> get(int playerIndex) {
        return Optional.ofNullable(tacticalSituations.get(playerIndex));
    }
}

File: src/main/java/tarehart/rlbot/planning/TacticsTelemetry.kt
package tarehart.rlbot.planning

import java.util.HashMap
import java.util.Optional

object TacticsTelemetry {

    private val tacticalSituations = HashMap<Int, TacticalSituation>()

    operator fun set(situation: TacticalSituation, playerIndex: Int) {
        tacticalSituations.put(playerIndex, situation)
    }

    fun reset(playerIndex: Int) {
        tacticalSituations.remove(playerIndex)
    }

    operator fun get(playerIndex: Int): TacticalSituation? {
        return tacticalSituations[playerIndex]
    }
}
